Feat/omar assign trainees

TryOmar opened the pull request.

TryOmar committed:
feat: implement assignMentor and unassignMentor actions with validation schema

m10090 committed:
refactor: update assignMentor and unassignMentor functions to use IDs instead of usernames, enhance validation, and improve error handling

TryOmar committed:
feat: add AssignMentorsPage and AssignMentorRow components for assigning mentors to trainees

TryOmar committed:
fix: correct indentation in assignMentor function and add Assign Mentors link to TrainingNavigation

m10090 committed:
feat: add revalidation for assignMentor and unassignMentor actions to refresh mentor assignment state

m10090 committed:
fix: update mentorId field to include references and ensure it is not nullable in Trainees schema

m10090 committed:
feat: enhance AssignMentorsPage and AssignMentorRow components with improved data fetching and error handling

m10090 committed:
feat: enhance assignMentor function to support reassigning deleted trainees and improve database queries

m10090 committed:
fix: update assignMentor function to handle reassigning deleted trainees and improve logic for trainee existence checks

m10090 commented on src/app/protected/trainings/[trainingId]/staff/mentors/actions/unassignMentor.ts:
```diff
@@ -0,0 +1,133 @@
+      }
+    }
+
+    // check if the trainee is assigned to the mentor in this training
```
this should be isn't assigned not `is assigned`

m10090 committed:
Refactor mentor assignment logic and update trainee handling

- Removed unnecessary checks for trainee existence in the trainees table during mentor assignment.
- Updated the logic to mark trainees as deleted instead of throwing errors when they are reassigned.
- Simplified the trainee insertion process when they do not exist in the trainees table.
- Commented out the MentorTraineeHistory schema definition for future reference.
- Changed the mentorId field in the Trainees schema to be non-nullable.
- Updated the journal metadata with a new tag and timestamp.

m10090 committed:
fix: streamline URL pattern definitions and enhance permission checks in training middleware

m10090 committed:
fix: update permissions for staff management and assign mentor routes to remove edit access

m10090 committed:
fix: correct URL pattern for assign-mentor route to assign-mentors
